### PR TITLE
Noe/transactional frames mobile

### PR DESCRIPTION
**Pull request by nmalzieu**
Full flow for tx frames, from connection to simulation
Some frames to test it with:

- you can test with that Base frame: https://base-frame-lyart.vercel.app/
- or many other frames to check out here => https://github.com/ephemeraHQ/converse-backend/blob/main/api/verifiedFrames/verifiedFrames.ts

🚧 This works with branch `kc/transaction-simulation-follow-up` of the backend, if https://github.com/ephemeraHQ/converse-backend/pull/54 is not merged yet

https://github.com/user-attachments/assets/807bc7c9-7627-41cd-b896-07a47836d31d

## Release Notes

- **New Features**
	- Introduced `CurrentAccount` component to display user account information.
	- Added `TransactionActions` component for managing transaction interactions.
	- Implemented `TransactionHeader` for enhanced transaction previews.
	- Added `simulateTransaction` function for transaction simulations.
	- Enhanced wallet management with `switchChain` and `sendTx` functions.
	- Introduced `TransactionContent` and `TransactionPreview` components for detailed transaction displays.
	- Added support for multiple blockchain chains in wallet connections.
	- Enhanced `Drawer` component with optional handle rendering.
	- Added `EmojiPicker` component with improved rendering behavior.

- **Improvements**
	- Enhanced error handling in transaction actions.
	- Improved text styling capabilities with new color options.
	- Streamlined transaction handling logic with new hooks.
	- Enhanced navigation capabilities by reinstating onboarding screens.

- **Translations**
	- Expanded localization support for transaction-related features.

- **Dependencies**
	- Updated several package dependencies for improved functionality.

### CODE REVIEW

File: utils/evm/external.ts [hunk 1]
```diff
@@ -1,73 +1,180 @@
...
+  const chain = useActiveWalletChain();
   const activeAccount = useActiveAccount();
   const activeWallet = useActiveWallet();
   const { disconnect } = useDisconnect();
+  const switchActiveWalletChain = useSwitchActiveWalletChain();
+  const wallets = useInstalledWallets();
+
+  useEffect(() => {
+    accountSingleton = activeAccount;
+  }, [activeAccount]);
```
[the remark this answers is not in the document]
Comment (nmalzieu) [reply utils/evm/external.ts:44]: goal of the singleton is to keep its value even if unmounts

File: components/TransactionPreview/TransactionPreview.tsx [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+  const simulateTx = useCallback(async () => {
+    if (transactionToPreview && address && !simulatingTx.current) {
+      simulatingTx.current = true;
+      try {
+        const simulationResult = await simulateTransaction(
+          account,
+          address,
+          transactionToPreview.chainId,
+          transactionToPreview
+        );
```
[the remark this answers is not in the document]
Comment (nmalzieu) [reply components/TransactionPreview/TransactionPreview.tsx:91]: there is a button to close, so no need

File: utils/frames.ts [hunk 3]
```diff
@@ -197,77 +199,52 @@ export const getFrameImage = (frame: FrameWithType) => {
...
+
+      // Let's display the transaction preview screen
+      const chainId = parseInt(txData.chainId.slice(7), 10);
+      const transactionData = {
+        id: uuidv4(),
+        chainId,
+        to: txData.params.to,
+        value: txData.params.value,
+        data: txData.params.data,
+      };
```
[the remark this answers is not in the document]
Comment (nmalzieu) [reply utils/frames.ts:235]: if the tx data is non well formatted, the simulation will fail and the user will know about it.

**Comment by fabriguespe** [comment]
hey @nmalzieu ! 

Updated Base Tx Frame
- https://github.com/ephemeraHQ/base-tx-frame

*For me its working in scenarios in Frames Debugger

## Testing

Here are the three testing URLs based on your requirements:

1. **Send 1 USDC to `0x73a333cb82862d4f66f0154229755b184fb4f5b0`:**

   ```
   https://base-tx-frame.vercel.app/transaction/?transaction_type=send&buttonName=Send&amount=1&token=USDC&receiver=0x73a333cb82862d4f66f0154229755b184fb4f5b0
   ```

   ```
   http://localhost:3000/transaction/?transaction_type=send&buttonName=Send&amount=1&token=USDC&receiver=0x73a333cb82862d4f66f0154229755b184fb4f5b0
   ```

2. **Swap 1 USDC to DAI:**

   ```
   https://base-tx-frame.vercel.app/transaction/?transaction_type=swap&buttonName=Swap&amount=1&token_from=USDC&token_to=DAI
   ```

   ```
   http://localhost:3000/transaction/?transaction_type=swap&buttonName=Swap&amount=1&token_from=USDC&token_to=DAI
   ```

3. **Mint the collection (with example of tokenId and collection):**

   ```
   https://base-tx-frame.vercel.app/transaction/?transaction_type=mint&buttonName=Mint&collection=0x73a333cb82862d4f66f0154229755b184fb4f5b0&token_id=1
   ```

   ```
   http://localhost:3000/transaction/?transaction_type=mint&buttonName=Mint&collection=0x73a333cb82862d4f66f0154229755b184fb4f5b0&token_id=1
   ```

File: components/Onboarding/ConnectViaWallet/ConnectViaWalletSupportedWallets.ts [hunk 4]
```diff
@@ -44,6 +67,36 @@ export const POPULAR_WALLETS = [
   },
 ];
 
+export const DEFAULT_SUPPORTED_CHAINS = [
```
Comment (alexrisch) [inline components/Onboarding/ConnectViaWallet/ConnectViaWalletSupportedWallets.ts:70]: Can we move this into a utils or constants file?
This way we can import it easier without also needing everything else in the file, long term it also avoids circular dependencies

File: components/TransactionPreview/TransactionContent.tsx [hunk 5]
```diff
@@ -0,0 +1,263 @@
...
+
+const TransactionResult = ({ status, error }: TransactionResultProps) => {
+  const { themed } = useAppTheme();
+  return (
+    <VStack style={status === "failure" ? themed($failure) : styles.center}>
+      {status === "failure" && (
+        <Text color="caution">{translate("transaction_failure")}</Text>
+      )}
+      <Text>
+        {status === "failure" ? error : translate("transaction_success")}
```
Comment (alexrisch) [inline components/TransactionPreview/TransactionContent.tsx:136]: I think this line needs a second look

File: components/TransactionPreview/TransactionContent.tsx [hunk 6]
```diff
@@ -0,0 +1,263 @@
...
+type ITransactionContentProps = {
+  simulation: SimulationState;
+  txState: TransactionState;
+  walletApp: InstalledWallet | undefined;
+  walletAddress: string | undefined;
+  address: string | undefined;
+  switchWallet: () => Promise<void>;
+};
+
+export const TransactionContent = ({
```
Comment (alexrisch) [inline components/TransactionPreview/TransactionContent.tsx:61]: Can you split this file into different components?

File: components/TransactionPreview/TransactionPreview.tsx [hunk 7]
```diff
@@ -0,0 +1,229 @@
...
+          });
+        } else {
+          // Let's refresh profile data for the addresses involved in the transaction
+          simulationResult?.changes?.forEach((change) => {
+            refreshProfileForAddress(account, change.to);
+          });
+          setSimulation({ status: "success", result: simulationResult });
+        }
+      } catch (e: any) {
+        console.log(e);
```
Comment (alexrisch) [inline components/TransactionPreview/TransactionPreview.tsx:105]: Logger instead

File: theme/colorsDark.ts [hunk 8]
```diff
@@ -49,6 +49,7 @@ export const colorsDark: IColors = {
     secondary: lightPalette.light60,
     tertiary: lightPalette.light30,
     inactive: lightPalette.light15,
+    caution: darkPalette.red,
```
Comment (alexrisch) [inline theme/colorsDark.ts:52]: Should this be lightPalette instead?

File: design-system/Text/Text.tsx [hunk 9]
```diff
@@ -10,6 +10,7 @@ export const Text = React.forwardRef<RNText, ITextProps>((props, ref) => {
   const {
     weight,
     size,
+    color,
```
Comment (alexrisch) [inline design-system/Text/Text.tsx:13]: cc @thierryskoda how would you like to handle these?
I think it's expected to instead be a preset? Or am I misunderstanding?

Comment (thierryskoda) [reply design-system/Text/Text.tsx:13]: I told Noe that it's okay for primary styling properties to also be props. Ideally we don't want too many and we should always look if the text we're adding is a preset but if not but we still need a custom color I think it's fine to have `weight` , `size` and  `color` as props

File: theme/colorsDark.ts (hunk 8 above)
Comment (thierryskoda) [reply theme/colorsDark.ts:52]: Probably my error but yes we should use lightPalette, even if it's the same color dark/light for now

**Review by alexrisch — APPROVED** [review]

**Review by alexrisch — APPROVED** [review]

**Pull request merged**